torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -2533,15 +2533,20 @@ def _sharded_pre_load_state_dict_hook(
 
             # All-gather the param (ShardedTensor)
             shards = param.local_shards()
-            local_tensor = cast(torch.Tensor, shards[0].tensor).flatten()
-            dim_0_size = param.size()[0]
             param_numel = param.size().numel()
+            dim_0_size = param.size()[0]
             chunk_size = (
                 math.ceil(dim_0_size / self.world_size) * param_numel // dim_0_size
             )
-            num_padding = chunk_size - local_tensor.numel()
-            if num_padding > 0:
-                local_tensor = F.pad(local_tensor, [0, num_padding])
+            if shards:
+                local_tensor = cast(torch.Tensor, shards[0].tensor).flatten()
+                if not local_tensor.is_cuda:
+                    local_tensor = local_tensor.cuda()
+                num_padding = chunk_size - local_tensor.numel()
+                if num_padding > 0:
+                    local_tensor = F.pad(local_tensor, [0, num_padding])
+            else:
+                local_tensor = torch.zeros(chunk_size, dtype=param.dtype).cuda()
```

Comment:
nit: May save an H2D copy
```suggestion
                local_tensor = torch.zeros(chunk_size, dtype=param.dtype, device=self.compute_device)
```
or `device=flat_param.device` (I do not think it should matter)